src/Lean/PrettyPrinter/Delaborator/Basic.lean

```diff
@@ -277,6 +277,17 @@ end Delaborator
 open SubExpr (Pos PosMap)
 open Delaborator (OptionsPerPos topDownAnalyze)
 
+/-- Custom version of `Lean.Core.betaReduce` to beta reduce expressions for the `pp.beta` option.
+We do not want to beta reduce the application in `let_fun` annotations. -/
+private partial def betaReduce' (e : Expr) : CoreM Expr :=
+  Core.transform e (pre := fun e => do
+    if isLetFun e then
+      return .done <| e.updateMData! (.app (← betaReduce' e.mdataExpr!.appFn!) (← betaReduce' e.mdataExpr!.appArg!))
```

Comment:
Mmh, is that different from `.continue`?

Reply:
`.continue` would visit the mdata value again, but here we want to skip visiting the `.app` itself and jump into its arguments. (It would be nice if `Core.transform` had a way to do these finer-grained recursions.)

Reply:
Yeah, "mdata value" was supposed to mean the `.app` itself, which is the immediate subterm of the mdata.